File: Assets/ToLua/Lua/user/ui/formation/model/formation_attr_data.lua
local t = {}
t.data = {}
local item = {}
item.__index = item

function item.New(table)
	local o = {}
	setmetatable(o,item)
	o.id = tonumber(table.id)
  o.type = tonumber(table.type)--1基础属性 2花钱激活属性
	o.comat_a = tonumber(table.comat_a)
	o.comat_b = tonumber(table.comat_b)
	o.time = tonumber(table.time)
	o.count = tonumber(table.counts)
	o.interval = tonumber(table.interval)
	o.targetType = tonumber(table.target)
	o.effectType = tonumber(table.effect_type)
	o.effect_base = tonumber(table.effect_base)
	o.effect_upgrade = tonumber(table.effect_upgrade)
	o.unlock_lv = tonumber(table.unlock_lv)
	return o
end
function item:GetEffectAttrValue(level)
  if self.type == 1 then
    return self.effect_base + (level - 1) * self.effect_upgrade
  else
    return self.effect_base
  end
  
end

function t.GetDataById(id)
	return t.data[id]
end
function t.GetFormationDatasByFormationId(formationId)
  local result = {}
  local index = 1
  for k,v in ipairs(t.data) do
    if v.id == formationId then
      result[index] = v
      index = index + 1
    end
  end
  return result
end

function t.GetFormationDatas(formationid,level)
  local result = {}
  local index = 1
  for k,v in ipairs(t.data) do
    if v.id == formationid and v.unlock_lv <= level then
      result[index] = v
      index = index + 1
    end
  end
  return result
end


local function Start()
	local origin = dofile('formation_attr')
	origin.ForEach(function(id,table)
		local newItem = item.New(table)
		t.data[id] = newItem
	end)
end

Start()
return t

File: Assets/Code/Logic/Formation/Model/FormationAttrData.cs
using System.Collections.Generic;
using Common.Util;
using Logic.Game.Model;
using Logic.Enums;

namespace Logic.Formation.Model
{
    public class FormationAttrData
    {
        private static List<FormationAttrData> _formationDataList;
        public static List<FormationAttrData> FormationDataList
        {
            get
            {
                if (_formationDataList == null)
                {
                    _formationDataList = CSVUtil.Parse<FormationAttrData>("config/csv/formation_attr");
                }
                return _formationDataList;
            }
        }

        public static List<FormationAttrData> GetFormationDatas(int id, FormationAttrType formationAttrType = FormationAttrType.Base)
        {
            List<FormationAttrData> dataList = new List<FormationAttrData>();
            FormationAttrData formationAttrData = null;
            for (int i = 0, count = FormationDataList.Count; i < count; i++)
            {
                formationAttrData = FormationDataList[i];
                if (formationAttrData.id == id && formationAttrType == formationAttrData.formationAttrType)
                {
                    dataList.Add(formationAttrData);
                }
            }
            return dataList;
        }

        public static List<FormationAttrData> GetFormationDatas(int id, int level, FormationAttrType formationAttrType = FormationAttrType.Base)
        {
            List<FormationAttrData> dataList = GetFormationDatas(id, formationAttrType);
            List<FormationAttrData> result = new List<FormationAttrData>();
            FormationAttrData formationAttrData = null;
            for (int i = 0, count = dataList.Count; i < count; i++)
            {
                formationAttrData = dataList[i];
                if (formationAttrData.unlock_lv <= level)
                {
                    result.Add(formationAttrData);
                }
            }
            return result;
        }


        [CSVElement("id")]
        public int id;

        public FormationAttrType formationAttrType;

        [CSVElement("type")]
        public int typeInt
        {
            set
            {
                formationAttrType = (FormationAttrType)value;
            }
        }

        [CSVElement("comat_a")]
        public float comat_a;//1级阵型，对队伍战力的系数

        [CSVElement("comat_b")]
        public float comat_b;//每升1级，增加的队伍战力系数

        [CSVElement("time")]
        public float time;

        [CSVElement("counts")]
        public uint count;

        [CSVElement("interval")]
        public int interval;

        public TargetType targetType;
        [CSVElement("target")]
        public int targetTypeInt
        {
            set
            {
                targetType = (TargetType)value;
            }
        }

        public FormationEffectType effectType;
        [CSVElement("effect_type")]
        public int effect_typeid
        {
            set
            {
                effectType = (FormationEffectType)value;
            }
        }
        [CSVElement("effect_base")]
        public float effect_base;

        [CSVElement("effect_upgrade")]
        public float effect_upgrade;

        [CSVElement("unlock_lv")]
        public int unlock_lv;



    }
}
